### PR TITLE
[CHK-3047] feat: added event info to v2 API + Unit tests

### PR DESCRIPTION
**Pull request by EmanueleBVtech**
#### List of Changes
- add `events info` list as new data returned from the v2 API 
- only `eventCode `and `creationDate` field are returned
- Added unit tests for the new `events info` list

#### Motivation and Context
The goal of this PR is to add a list of events to searchTransaction v2 api. Only `eventCode `and `creationDate` fields must return of the original `Event` structure from `eventstore`

#### How Has This Been Tested?

Unit test

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/main/kotlin/it/pagopa/ecommerce/helpdesk/dataproviders/v2/mongo/EcommerceTransactionDataProvider.kt [hunk 1]
```diff
@@ -144,8 +144,13 @@ class EcommerceTransactionDataProvider(
                         },
                         ::Pair
                     )
-                    .map { (baseTransaction, email) ->
-                        baseTransactionToTransactionInfoDtoV1(baseTransaction, email)
+                    .zipWith(events.collectList(), ::Pair)
+                    .map { (baseTransactionEmail, events) ->
```
Comment (infantesimone) [inline src/main/kotlin/it/pagopa/ecommerce/helpdesk/dataproviders/v2/mongo/EcommerceTransactionDataProvider.kt:148]: variable names are one of the most difficult choices for a developer :-) maybe something like `baseTransactionAndEmailPair`?

Comment (EmanueleBVtech) [reply src/main/kotlin/it/pagopa/ecommerce/helpdesk/dataproviders/v2/mongo/EcommerceTransactionDataProvider.kt]: done

File: src/main/kotlin/it/pagopa/ecommerce/helpdesk/dataproviders/v2/mongo/EcommerceTransactionDataProvider.kt [hunk 2]
```diff
@@ -144,8 +144,13 @@ class EcommerceTransactionDataProvider(
...
                     )
-                    .map { (baseTransaction, email) ->
-                        baseTransactionToTransactionInfoDtoV1(baseTransaction, email)
+                    .zipWith(events.collectList(), ::Pair)
+                    .map { (baseTransactionAndEmailPair, events) ->
+                        baseTransactionToTransactionInfoDtoV1(
+                            baseTransactionAndEmailPair.first,
+                            baseTransactionAndEmailPair.second,
+                            events
+                        )
```
Comment (pietro-tota) [inline src/main/kotlin/it/pagopa/ecommerce/helpdesk/dataproviders/v2/mongo/EcommerceTransactionDataProvider.kt:153]: what about mapping the above Pair to a Triple and use arguments deconstructing for making code clearer?
Something like:
```suggestion
                     .flatMap { (baseTransaction, email) ->
                        events.collectList().map { Triple(baseTransaction, email, it) }
                    }
                    .map { (baseTransaction, email, events) ->
                        baseTransactionToTransactionInfoDtoV1(baseTransaction, email, events)
                    }
```

in this way we can avoid calling `first` and `second` method on the pair class and use the Triple argument deconstructing making code easier to refactoring and removing the variable name baseTransactionAndEmailPair / first and second methods link

File: src/main/kotlin/it/pagopa/ecommerce/helpdesk/dataproviders/v2/mongo/EcommerceTransactionDataProvider.kt [hunk 3]
```diff
@@ -173,8 +178,13 @@ class EcommerceTransactionDataProvider(
                         },
                         ::Pair
                     )
-                    .map { (baseTransaction, email) ->
-                        baseTransactionToTransactionInfoDtoV2(baseTransaction, email)
+                    .zipWith(events.collectList(), ::Pair)
```
Comment (pietro-tota) [inline src/main/kotlin/it/pagopa/ecommerce/helpdesk/dataproviders/v2/mongo/EcommerceTransactionDataProvider.kt:181]: same as above, what do you think?

**Review by pietro-tota — DISMISSED** [review]
Well done! Just a trivial issue 🚀

**Review by pietro-tota — APPROVED** [review]
Well done!

**Review by infantesimone — APPROVED** [review]

**Pull request merged**